configuration-api/main/advancedAuth.js

```diff
@@ -27,11 +36,28 @@ const typeDefs = gql`
     language: String!
     lastTenantSelected: String
   }
+  type ResourceType {
+    name: String!
+    userID: String!
+    tenantID: String!
+  }
+  type ResourceEndpoint {
+    name: String!
+    resourceTypeName: String!
+    nameAndID: String!
+  }
   type Query {
     listTenants(tenantNames: [String]!): [TenantConfiguration]
     getUserPreferences(userName: String!): [UserPreferencies]
+    getUserResourceType(tenantID: String!): [ResourceType]
+    getEndpoints(resourceTypeName: String!): [ResourceEndpoint]
   }
   type Mutation {
+    newResourceType(name: String!, userID: String!, tenantID: String!): [ResourceType]
+    deleteResourceType(name: [String]!): [ResourceType]
+    deleteThisEndpoint(name: [String]!): [ResourceEndpoint]
```

Comment:
why is the end point separated from the resource???

Reply:
we have discussed that, is for a more controllability but is linked by a JOIN to the resource, i would prefer to keep them separated because is more future proof if we need to change something related to the logic

Reply:
still, this should be then linked to the resourceType, where is the link?

Reply:
> still, this should be then linked to the resourceType, where is the link?

RESOURCE
{"id":"58afe6c3-5214-4876-adce-1dd0a6721300/resourceName","name":"resourceName","userID":"admin@mail.com","tenantID":"58afe6c3-5214-4876-adce-1dd0a6721300"}

ENDPOINT
{"nameAndID":"resourceName/58afe6c3-5214-4876-adce-1dd0a6721300","name":"http://localhost:1026/v2/entities","resourceTypeName":"resourceName"}

Reply:
> why are you combining the id with the resource name? let's keep things clean and consistent.
> 
> 1. id is made by the id only, you don't need the name in it.
> 2. links are over id, not over name.

if a resource has the same link and the same resource linked it won't work

Reply:
> > why are you combining the id with the resource name? let's keep things clean and consistent.
> > 
> > 1. id is made by the id only, you don't need the name in it.
> > 2. links are over id, not over name.
> 
> if a resource has the same link and the same resource linked it won't work

how can you differentiate a resource with the same name but binded in 2 different tenants if the id is not the tenantID/name?

Reply:
> before inserting can't you check if the value exist?

i can but everything is automated if i put a field inside the mongo schema with unique:true, otherwise i need to to a query a control on the length and return of the error and i think that is not good for the performance and also for the readability of the code